sdk/lib/opentelemetry/sdk/configurator.rb

```diff
@@ -143,19 +143,42 @@ def install_instrumentation
       end
 
       def configure_span_processors
-        processors = @span_processors.empty? ? [default_span_processor] : @span_processors
+        processors = @span_processors.empty? ? [wrapped_exporter_from_env] : @span_processors
         processors.each { |p| tracer_provider.add_span_processor(p) }
       end
 
-      def default_span_processor
-        Trace::Export::SimpleSpanProcessor.new(
-          Trace::Export::ConsoleSpanExporter.new
-        )
+      def wrapped_exporter_from_env
+        exporter = ENV.fetch('OTEL_TRACES_EXPORTER', 'otlp')
+        case exporter
+        when 'otlp' then fetch_exporter(exporter, 'OpenTelemetry::Exporter::OTLP::Exporter')
+        when 'jaeger' then fetch_exporter(exporter, 'OpenTelemetry::Exporter::Jaeger::CollectorExporter')
+        when 'zipkin' then fetch_exporter(exporter, 'OpenTelemetry::Exporter::Zipkin::Exporter')
+        else
+          OpenTelemetry.logger.warn "The #{exporter} exporter is unknown and cannot be configured, spans will not be exported"
+          Trace::NoopSpanProcessor.instance
+        end
       end
 
-      def configure_propagation
-        OpenTelemetry.propagation = create_propagator(@injectors || default_injectors,
-                                                      @extractors || default_extractors)
+      def configure_propagation # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity
+        propagators = ENV.fetch('OTEL_PROPAGATORS', 'tracecontext,baggage').split(',')
+        injectors, extractors = propagators.uniq.collect do |propagator|
+          case propagator
+          when 'tracecontext'
+            [OpenTelemetry::Trace::Propagation::TraceContext.text_map_injector, OpenTelemetry::Trace::Propagation::TraceContext.text_map_extractor]
+          when 'baggage'
+            [OpenTelemetry::Baggage::Propagation.text_map_injector, OpenTelemetry::Baggage::Propagation.text_map_extractor]
+          when 'b3' then fetch_propagator(propagator, 'OpenTelemetry::Propagator::B3::Single')
+          when 'b3multi' then fetch_propagator(propagator, 'OpenTelemetry::Propagator::B3::Multi', 'b3')
+          when 'jaeger' then fetch_propagator(propagator, 'OpenTelemetry::Propagator::Jaeger')
+          when 'xray' then fetch_propagator(propagator, 'OpenTelemetry::Propagator::XRay')
```

Comment:
there's probably a place in the future where this lives in a `-contrib` repo with other cloud provider (or vendor 😅 ) specific formats and we need some different handling here, but for now i think this makes sense

Reply:
Yeah, maybe - for some (political?) reason, this is required by the spec 🤷